src/test_typing_extensions.py

```diff
@@ -1879,6 +1879,24 @@ class C: pass
         P.__init__(c, 1)
         self.assertEqual(c.x, 1)
 
+    @only_with_typing_Protocol
+    def test_protocol_defining_init_does_not_get_overridden_2(self):
```

Comment:
Does this work if the positions of `typing.Protocol` and `typing_extensions.Protocol` in the MRO are reversed?

Reply:
Yup! Added some tests for that in e68be1a